neuro_san/http_sidecar/interfaces/agent_authorizer.py

```diff
@@ -0,0 +1,28 @@
+
+# Copyright (C) 2023-2025 Cognizant Digital Business, Evolutionary AI.
+# All Rights Reserved.
+# Issued under the Academic Public License.
+#
+# You can be released from the terms, and requirements of the Academic Public
+# License by purchasing a commercial license.
+# Purchase of a commercial license is mandatory for any use of the
+# neuro-san SDK Software in commercial settings.
+#
+# END COPYRIGHT
+"""
+See class comment for details
+"""
+
```

Comment:
Factor out an abstract interface for deciding - will we allow API request to go in for this particular agent at this point?

Reply:
Very interesting.  Might be able to expand this for other uses in the future.
Also future: Should grpc side also use this interface?